api/README.md

```diff
@@ -0,0 +1,11 @@
+API text files
+==============
```

Comment:
I prefer that documentation is consistent.
All `.md` files in the repo uses the GH flavor, e.g. https://github.com/rust-bitcoin/rust-bitcoin/blob/a9cdcb439291a0ad3b67edfcd76de8fd2800aca4/CONTRIBUTING.md?plain=1#L1-L5

```suggestion
# API text files
```

Reply:
Ah yeah, sure thing. Thanks!

Reply:
I tend to use `====` in my own stuff so I went to check that I hadn't introduced any other files and there are tons in this repo
```
gg '===='
base58/README.md:2:  =======================
bitcoin/tests/data/README.md:2:  ================
bitcoin/tests/data/serde/README.md:2:  ==========================
fuzz/README.md:103:=====================================================================
internals/README.md:2:  ======================
io/README.md:2:  =======================
units/README.md:2:  =============
```

We should probably make them all uniform.

Reply:
`gg` is an alias to `git grep -IPn --color=always`

Reply:
Yes my

> All .md files

was based on a sample from the root dir `.md`s. I should have been more specific :/

We should definitely normalize all of them.
I can work in a future PR to normalize markdown, I am quite used from an old gig that I was in charge of technical documentation in scientific software.

Reply:
Sweet, nice one.